doc/adr/0010-atomic-composite-grapth-operations.md

```diff
@@ -0,0 +1,267 @@
+# 10. Atomic/Composite Grapth Operations
+
+Date: 2021-08-18
+
+## Status
+
+Draft
+
+Supercedes [7. Lan-Party Datascript Events](0007-lan-party-datascript-events.md)
+
+## Context
+
+We've made an effort to support remote execution of Semantic Events.
+
+These events where direct port of `events.cljs` which where mostly informed by UI concerns.
+
+Result is that we have a lot of different events that are doing same atomic operations over and over,
+but are not really reusing these Atomic Ops.
+
+Implementing `:block/save` that sometimes is just updating `:block/string` and other times also needs to `:page/create`.
+`:paste` event is another that will be super hard to implement w/o Atomic Graph Operations.
+
+
+## Decision
+
+We have two kinds of events to modify graph:
+- ⚛️ Atomic Graph Ops
+    - Not divisible Graph Ops
+    - Operations like create new block, create page, save block
+- ⎄ Composite Graph Ops
+    - Collection of events to be executed on the graph
+    - Like `:block/save` when new link is discovered, should produce also `:page/create` event
+
+Atomic events should follow same Common Events model as it's happening now.
+
+List of ((⚛️ Atomic Graph Ops))
+- *Page Ops*
+    - ((⚛️**`:page/new`**))
+    - ((⚛️**`:page/rename`**))
+    - ((⚛️**`:page/merge`**))
+    - ((⚛️**`:page/remove`**))
+- *Block Ops*
+    - ((⚛️**`:block/new`**))
+    - ((⚛️**`:block/save`**))
+    - ((⚛️**`:block/open`**))
+    - ((⚛️**`:block/move`** ⭐️))
+    - ⚛️**`:block/remove`**
+- *Shortcut Ops*
+    - (({{[[TODO]]}} ⚛️**`:shortcut/new`**))
+    - (({{[[TODO]]}} ⚛️**`:shortcut/remove`**))
+    - (({{[[TODO]]}} ⚛️**`:shortcut/move`**))
+
+
+((⎄ Composite Graph Ops)) are a way to group ((⚛️ Atomic Graph Ops)).
+
+We'll only need one type of Composite Graph Ops, that is Consequence Event.
+
+Consequence Event is 2 things:
+- Trigger Event (like `:paste`)
+- Consequence Events List
```

Comment:
Nit: it's a bit weird to have names like `trigger` and `consequence` to describe what's just a nested tree structure.

Reply:
Please propose better names.